### ISSUE TITLE
Composer uses wrong permissions when extracting zip file

### ISSUE THREAD

**Issue by a-camacho** [composer/composer#8556]
My `composer.json`:

```json
{
    "name": "ebea/wp-core-skeleton",
    "type": "project",
    "description": "Wordpress composer skeleton for EBEA websites.",
    "require": {
        "php": ">=7.0",
	    "composer/installers": "~1.0",
        "johnpbloch/wordpress-core-installer": "^1.0",
        "johnpbloch/wordpress-core": "*",
        "ebea/wp-layout-mosaik": "^0.1.2"
    },
    "repositories": [{
        "type": "composer",
        "url": "https://repository.ebea.io/"
    }],
    "extra": {
        "installer-paths" : {
            "wp-content/plugins/{$name}/" : ["type:wordpress-plugin"],
            "wp-content/themes/{$name}/" : ["type:wordpress-theme"]
        },
        "wordpress-install-dir": "wp"
    },
    "scripts": {
        "install-wp": "/home/tools/wp-deployer/scripts/post-install.sh",
        "post-install-fix-permissions": "/home/tools/wp-deployer/scripts/post-install-fix-wp-permissions.sh",
	    "import-wp-content": "/home/tools/wp-deployer/scripts/import-content.sh",
	    "import-wp-settings": "/home/tools/wp-deployer/scripts/import-settings.sh",
	    "import-wp-widgets": "/home/tools/wp-deployer/scripts/import-widgets.sh",
        "crontab-exc": "/home/tools/wp-deployer/scripts/crontab_exec.sh",
        "composer-to-json": "/home/tools/wp-deployer/scripts/composerToJson.sh"
    }
}
```

Output of `composer diagnose`:

```
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: fb93036a70c9520cab99b46723ebc8f2e6fd82a6
PHP version: 7.2.27
PHP binary path: /usr/bin/php7.2
OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019
```

When I run this command:

```
composer create-project ebea/wp-core-skeleton {{folder_name}} --repository=https://repository.entreprise.us
```

I get the following output:

```
Installing ebea/wp-core-skeleton (v0.4.1)
  - Installing ebea/wp-core-skeleton (v0.4.1): Downloading (100%)         
Created project in andre.temp
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 3 installs, 0 updates, 0 removals
  - Installing composer/installers (v1.7.0): Downloading (100%)         
  - Installing johnpbloch/wordpress-core-installer (1.0.2): Downloading (100%)         
  - Installing johnpbloch/wordpress-core (5.3.2): Downloading (100%)         
Writing lock file
Generating autoload files
```

And I expected this to happen:
- I expected Composer to install everything at the right place, which it does.
- BUT I also expected Composer to use traditionnal CHMOD (from server) when unziping .zip files from repository. Unfortunately, Composer gives my files and directories these CHMOD values :

<img width="563" alt="image" src="https://user-images.githubusercontent.com/29976133/73346905-f6176280-4286-11ea-810f-bbf1cd4338b9.png">

<img width="591" alt="iamge2" src="https://user-images.githubusercontent.com/29976133/73347336-b69d4600-4287-11ea-92d4-39aadaca3860.png">

I also openened an issue on Satis side, but I've been told that it doesn't have much to do with Satis : https://github.com/composer/satis/issues/577. I also found this issue that has been closed, and can't understand if I fall under this issue : https://github.com/composer/satis/issues/314

Thank you very much !

**Comment by alcohol** [comment]
Please search for similar issues reported already.

**Comment by Seldaek** [comment]
I believe this will be fixed by 1.10, but will require you to rebuild the archives.
